snuba/migrations/groups.py

```diff
@@ -257,25 +254,68 @@ def get_migrations(self) -> Sequence[str]:
         return ["0001_functions"]
 
 
-class GenericMetricsLoader(DirectoryLoader):
-    def __init__(self) -> None:
-        super().__init__("snuba.migrations.snuba_migrations.generic_metrics")
+HARDCODED_MIGRATIONS = {
+    "SYSTEM": "system",
+    "EVENTS": "events",
+    "TRANSACTIONS": "transactions",
+    "DISCOVER": "discover",
+    "OUTCOMES": "outcomes",
+    "METRICS": "metrics",
+    "SESSIONS": "sessions",
+    "QUERYLOG": "querylog",
+    "PROFILES": "profiles",
+    "FUNCTIONS": "functions",
+    "REPLAYS": "replays",
+}
 
-    def get_migrations(self) -> Sequence[str]:
-        return [
-            "0001_sets_aggregate_table",
-            "0002_sets_raw_table",
-            "0003_sets_mv",
-            "0004_sets_raw_add_granularities",
-            "0005_sets_replace_mv",
-            "0006_sets_raw_add_granularities_dist_table",
-            "0007_distributions_aggregate_table",
-            "0008_distributions_raw_table",
-            "0009_distributions_mv",
-        ]
+CONFIG_BUILT_MIGRATIONS = {
+    loader.migration_group_name: loader
+    for loader in [
+        ConfigurationLoader(config_file)
+        for config_file in glob(settings.MIGRATION_CONFIG_FILES_GLOB, recursive=True)
+    ]
+}
 
 
-_REGISTERED_GROUPS = {
+class _MigrationGroup(type):
```

Comment:
Why does this need to be a metaclass?

Reply:
Because if I don't make it a metaclass, `__getattr__` will only work on instances of `MigrationGroup`, instead of on the class itself.

Reply:
Damn, we really getting funky now! Love it